src/monmove.cpp

```diff
@@ -683,7 +686,65 @@ void monster::move()
                 // Friendly fire and pushing are always bad choices - they take a lot of time
                 bad_choice = true;
             }
+            //Shove vehicles out of way
+            if( this->has_flag( MF_PUSH_VEH ) ) {
+                auto vp = g->m.veh_at( candidate );
+                if( vp ) {
+                    bool move_veh = false;
+                    bool rotate_veh = false;
+                    bool damage_veh = false;
+                    vehicle &veh = vp->vehicle();
+                    units::mass veh_mass = veh.total_mass();
+                    switch( this->get_size() ) {
+                        case MS_TINY:
+                        case MS_SMALL:
+                            break;
+                        case MS_MEDIUM:
+                            if( veh_mass < 5000_kilogram ) {
+                                move_veh = true;
+                                rotate_veh = true;
+                                damage_veh = true;
+                            }
+                            break;
+                        case MS_LARGE:
+                            if( veh_mass < 25000_kilogram ) {
+                                move_veh = true;
+                                rotate_veh = true;
+                                damage_veh = true;
+                            }
+                            break;
+                        case MS_HUGE:
+                            if( veh_mass < 500000_kilogram ) {
+                                move_veh = true;
+                                rotate_veh = true;
+                                damage_veh = true;
+                            }
+                            break;
+                        default:
+                            break;
+                    }
+                    if( move_veh ) {
+                        const int dpx = clamp( destination.x - candidate.x, -1, 1 );
+                        const int dpy = clamp( destination.y - candidate.y, -1, 1 );
+                        const int dpz = clamp( destination.z - candidate.z, -1, 1 );
+                        tripoint dp = { dpx, dpy, dpz };
+                        if( dp != tripoint_zero ) {
+                            if( damage_veh ) {
+                                veh.smash( 0.01F, 0.1F, 0.2F );
```

Comment:
You could also use `damage_origin` (would be the part closest to the monster). So that they're less likely to break the parts they can't really reach.

Reply:
Didn't implement it yet as I have troubles with coordinates, but will address this later.